o Add the "broken_shadow" argument to pam_unix, for ignoring errors reading
  shadow information (for example, when it doesn't exist due to your NIS maps
  being screwed up).
--- Linux-PAM-0.77/modules/pam_unix/README	2002-05-29 01:31:12.000000000 -0400
+++ Linux-PAM-0.77/modules/pam_unix/README	2003-06-27 19:21:13.000000000 -0400
@@ -31,5 +31,7 @@
 	nis		-	use NIS RPC for setting new password
 	remember=X	-	remember X old passwords, they are kept in
 				/etc/security/opasswd in MD5 crypted form
+	broken_shadow	-	ignore errors reading shadow information for
+				users in the account management module
 
 	invalid arguments are logged to syslog.
--- Linux-PAM-0.77/modules/pam_unix/support.h	2002-07-11 01:43:51.000000000 -0400
+++ Linux-PAM-0.77/modules/pam_unix/support.h	2003-06-27 19:23:39.000000000 -0400
@@ -81,8 +81,10 @@
 #define UNIX_LIKE_AUTH           19	/* need to auth for setcred to work */
 #define UNIX_REMEMBER_PASSWD     20	/* Remember N previous passwords */
 #define UNIX_NOREAP              21     /* don't reap child process */
+#define UNIX_BROKEN_SHADOW       22     /* ignore errors reading password aging
+					 * information during acct management */
 /* -------------- */
-#define UNIX_CTRLS_              22	/* number of ctrl arguments defined */
+#define UNIX_CTRLS_              23	/* number of ctrl arguments defined */
 
 
 static const UNIX_Ctrls unix_args[UNIX_CTRLS_] =
@@ -112,6 +114,7 @@
 /* UNIX_LIKE_AUTH */       {"likeauth",        _ALL_ON_,            01000000},
 /* UNIX_REMEMBER_PASSWD */ {"remember=",       _ALL_ON_,            02000000},
 /* UNIX_NOREAP */          {"noreap",          _ALL_ON_,            04000000},
+/* UNIX_BROKEN_SHADOW */   {"broken_shadow",   _ALL_ON_,           010000000},
 };
 
 #define UNIX_DEFAULTS  (unix_args[UNIX__NONULL].flag)
--- Linux-PAM-0.77/modules/pam_unix/pam_unix_acct.c	2003-06-27 19:21:13.000000000 -0400
+++ Linux-PAM-0.77/modules/pam_unix/pam_unix_acct.c	2003-06-27 19:21:13.000000000 -0400
@@ -129,6 +129,10 @@
 	}
 
 	if (!spent)
+		if (ctrl & UNIX_BROKEN_SHADOW)
+			return PAM_SUCCESS;
+
+	if (!spent)
 		return PAM_AUTHINFO_UNAVAIL;	/* Couldn't get username from shadow */
 
 	curdays = time(NULL) / (60 * 60 * 24);
